File: typescript/src/workshop/m5/s05_extractsubclass/step3/Programme.ts
import { Screening } from './Screening.js';

/** Krok 3: bez zmian. */
export class Programme {
  line(title: string, format: string, guest: string | null): string {
    const screening: Screening = guest === null
      ? Screening.regular(title, format)
      : Screening.premiere(title, format, guest);
    return screening.describe() + ' | ' + screening.price();
  }
}

File: csharp/src/Training.Workshop/M5/S05ExtractSubclass/Step3/Programme.cs
namespace Training.Workshop.M5.S05ExtractSubclass.Step3;

/// <summary>Krok 3: bez zmian.</summary>
public sealed class Programme
{
    public string Line(string title, string format, string? guest)
    {
        var screening = guest == null
            ? Screening.Regular(title, format)
            : Screening.Premiere(title, format, guest);
        return screening.Describe() + " | " + screening.Price();
    }
}
